### PR TITLE
Move WF/ConstEvaluatable goal to clause

### PR DESCRIPTION
**Pull request by compiler-errors**
It can show up in a param-env, so I think it needs to be a clause kind.

r? @lcnr or @oli-obk

### CODE REVIEW

**Comment by compiler-errors** [comment]
astconv is too large :facepalm:

File: compiler/rustc_lint/src/builtin.rs [hunk 1]
```diff
@@ -1611,12 +1611,12 @@ impl<'tcx> LateLintPass<'tcx> for TrivialConstraints {
                     AliasRelate(..) |
                     // Ignore bounds that a user can't type
                     Clause(Clause::WellFormed(..)) |
+                    Clause(Clause::ConstEvaluatable(..)) |
                     ObjectSafe(..) |
                     ClosureKind(..) |
                     Subtype(..) |
                     Coerce(..) |
                     // FIXME(generic_const_exprs): `ConstEvaluatable` can be written
```
Comment (oli-obk) [inline compiler/rustc_lint/src/builtin.rs:1619]: this needs to be moved up, too

**Review by oli-obk — CHANGES_REQUESTED** [review]
r=me with tidy happy

**Comment by compiler-errors** [comment]
Moved some files around astconv to appease tidy, since that file is huge.

r? @oli-obk @bors r=oli-obk

**Comment by matthiaskrgr** [comment]
@bors r-
maybe needs rebase?  https://github.com/rust-lang/rust/actions/runs/5299075781/jobs/9591818074#step:25:1355

**Comment by compiler-errors** [comment]
Yeah, conflicted with another pr.

@bors r=lcnr

**Pull request merged**